pkg/reconciler/route/traffic/rollout.go

```diff
@@ -0,0 +1,63 @@
+/*
+Copyright 2020 The Knative Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+// rollout.go contains the types and functions to deal with
+// gradual rollout of the new revision for a configuration target.
+// The types in this file are expected to be serialized as strings
+// and used as annotations for progressive rollout logic.
+
+package traffic
+
+// RolloutAnnotationKey is the annotation key for storing
+// the rollout state in the Annotations of the Kingress or Route.Status.
+const RolloutAnnotationKey = "networking.intenal.knative.dev/rollout"
+
+// Rollout encapsulates the current rollout state of the system.
```

Comment:
Suppose we want to make changes here.  How do we version it?  What if we rollout a new controller in the middle of a rollout?  might be a nice upgrade test 😉

Reply:
Yeah, it'll be interesting.
In general I expect standard API type additions/deprecations — new fields are all optional, old fields first stop being read, then in a version stop being written, then removed.